dspace-server-webapp/src/test/java/org/dspace/app/rest/matcher/AccessConditionOptionMatcher.java

```diff
@@ -35,11 +41,54 @@ public static Matcher<? super Object> matchAccessConditionOption(String name,
                                               : hasNoJsonPath("$.hasStartDate"),
                 Objects.nonNull(hasEndDate) ? hasJsonPath("$.hasEndDate", is(hasEndDate))
                                             : hasNoJsonPath("$.hasEndDate"),
-                StringUtils.isNotBlank(maxStartDate) ? hasJsonPath("$.maxStartDate", notNullValue())
+                StringUtils.isNotBlank(maxStartDate) ? hasJsonPath("$.maxStartDate", new DateMathMatcher(maxStartDate))
                                                      : hasNoJsonPath("$.maxStartDate"),
-                StringUtils.isNotBlank(maxEndDate) ? hasJsonPath("$.maxEndDate", notNullValue())
+                StringUtils.isNotBlank(maxEndDate) ? hasJsonPath("$.maxEndDate", new DateMathMatcher(maxEndDate))
                                                      : hasNoJsonPath("$.maxEndDate")
                 );
     }
 
+    /**
+     * Internal matcher to compare an ISO date from JSON with a DateMath expression.
+     */
+    private static class DateMathMatcher extends BaseMatcher<String> {
```

Comment:
Is there any case where this matcher is used to compate `Date`s to date expressions like `+6MONTHS`?
I didn't find one; we can probably get away with just a simple `is()` comparison on the date expressions.